src/modules/app/appSaga.ts

```diff
@@ -507,5 +552,7 @@ export function* appSaga() {
   yield takeLatest(DOWNLOAD_CODE_SNIPPET, downloadCodeSnippet);
   yield takeLatest(UPDATE_VISUALIZATION, updateVisualizationType);
   yield takeLatest(COPY_API_RESOURCE_URL, copyApiResourceUrl);
+  yield takeLatest(SAVE_EXISTING_QUERY, saveExistingQuery);
```

Comment:
IMO we should not creating separate action for `saving existing query` - as later in the UI the devs will need to consider this both scenarios. 

There should be only one action responsible for **Saving Query** and the different flows should be executed inside saga based on conditions (exist or not exist).